### PR TITLE
Fix mutation codegen

### PR DESCRIPTION
**Pull request by r52**
Fix `copyWith(MutationState)` type and add missing exports

### CODE REVIEW

File: packages/riverpod_annotation/lib/riverpod_annotation.dart [hunk 1]
```diff
@@ -25,6 +25,13 @@ export 'package:riverpod/src/internals.dart'
         $RefArg,
         $ProviderPointer,
 
+        // Mutation/Listenables
+        ProviderListenable,
+        LazyProxyListenable,
```
Comment (rrousselGit) [inline packages/riverpod_annotation/lib/riverpod_annotation.dart:30]: Those aren't public APIs. We should rename those we $ if we want to show them to the generator

File: packages/riverpod_generator/lib/src/templates/mutation.dart [hunk 2]
```diff
@@ -67,7 +67,7 @@ final class ${mutation.generatedMutationImplName}
   }
 
   @override
-  ${mutation.generatedMutationImplName} copyWith(MutationState<int> state, {Object? key}) => ${mutation.generatedMutationImplName}(element, state: state, key: key);
+  ${mutation.generatedMutationImplName} copyWith(MutationState<${provider.valueTypeDisplayString}> state, {Object? key}) => ${mutation.generatedMutationImplName}(element, state: state, key: key);
```
Comment (rrousselGit) [inline packages/riverpod_generator/lib/src/templates/mutation.dart:70]: Oopsy!

Can you add tests for those?

Comment (r52) [reply packages/riverpod_generator/lib/src/templates/mutation.dart:70]: Test added

File: packages/riverpod_annotation/lib/riverpod_annotation.dart (hunk 1 above)
Comment (r52) [reply packages/riverpod_annotation/lib/riverpod_annotation.dart:30]: $LazyProxyListenable seems fine, but ProviderListenable looks like its used in a lot of places that seems like it needs to be resolvable publicly, and I don't have enough experience to tell whether they can simply all be safely renamed. Your thoughts?

**Comment by rrousselGit** [comment]
LGTM, thanks!

**Pull request merged**